frontend/public/components/utils/horizontal-nav.tsx

```diff
@@ -274,18 +273,19 @@ export const HorizontalNav = React.memo((props: HorizontalNavProps) => {
 
   const routes = pages.map((p) => {
     const path = `${props.match.path}/${p.path || p.href}`;
-    const render = (params) => {
+    const render = (params: RouteComponentProps) => {
       return (
         <p.component
           {...componentProps}
           {...extraResources}
           {...p.pageData}
-          params={params}
+          {..._.omit(params, 'match')} // match is being pushed as part of componentProps
+          params={params} //Todo: This should be removed
```

Comment:
Details page uses `HorizontalNav`. More than 200 components have access to all these props. Keeping this as of now to not break any of those components.